cli/internal/kubernetes/upgrade.go

```diff
@@ -354,6 +355,18 @@ func (u *Upgrader) GetClusterAttestationConfig(ctx context.Context, variant vari
 	return existingAttestationConfig, existingConf, nil
 }
 
+// BackupConfigMap creates a backup of the given config map.
+func (u *Upgrader) BackupConfigMap(ctx context.Context, cm *corev1.ConfigMap) error {
```

Comment:
This is more of a "CreateConfigMap" if the caller is responsible for setting up the Config Map to create.
I think we can reduce actions required by the caller if this function simply takes a string, the name of the Config Map, and handles fetching the map and creating the backup from that.